### ISSUE TITLE
页面中引入一个同名组件，看上去被src/components中的组件覆盖

### ISSUE THREAD

**Issue by quixote1304629** [unplugin/unplugin-vue-components#717]
### Describe the bug

**库版本**
vue2.7
unplugin-vue-components/webpack

**组件层级**
src
--componnets
----A.vue
--views
----OnePageView.vue
----A.vue

**OnePageView.vue**
`
<template>
  <div class="hoopuu">
    <A />
  </div>
</template>

<script>
import A from './A.vue'
export default {
 components: {A},
  data() {
    return {
      text: 'hoopuu123',
    };
  },
};
</script>
`

**问题描述**
1. src/components 有被配置到dirs属性中
2. OnePageView.vue中渲染的组件理论上期望是 ./A.vue, 但实际上渲染的组件是src/components/A.vue。
3. 我用的是webpack

有没有大佬在遇到类似组件同名的问题

### Reproduction

略

### System Info

```Shell
略
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by k644606347** [comment]
这个问题有进展吗？我也遇到了，我是rspack

**Comment by ilyaliao** [comment]
There’s a workaround by setting `allowOverrides: true`, but I wouldn’t recommend it since, AFAIK, the resolution order isn’t guaranteed.

I’d suggest avoiding duplicate component names altogether, or moving `components/HelloWorld.vue` into a subfolder (I noticed you enabled `directoryAsNamespace`).

You can also work around it like this:

```ts
Components({
  dirs: ['src'],
  directoryAsNamespace: true,
  globalNamespaces: ['components'],
  dts: true,
}),
```

Then you can just use it as `ViewsHelloWorld`.

**Comment by ilyaliao** [comment]
Alternatively, maybe we should skip replacing components that are already explicitly imported.

**Comment by k644606347** [comment]
> Alternatively, maybe we should skip replacing components that are already explicitly imported.

Indeed, this is in line with intuition. Can improvements be made in the future? This is very valuable because this plugin is very popular; Perhaps it is difficult to implement because the plugin needs to be compatible with a range of build tools such as Vite/Webpack; We also need to consider handling the. vue/. tsx encoding methods;

**Comment by ilyaliao** [comment]
By using the Composition API, you can avoid this issue. Would you be open to giving it a try?

**Comment by k644606347** [comment]
> > > Alternatively, maybe we should skip replacing components that are already explicitly imported.
> > 
> > 
> > Indeed, this is in line with intuition. Can improvements be made in the future? This is very valuable because this plugin is very popular; Perhaps it is difficult to implement because the plugin needs to be compatible with a range of build tools such as Vite/Webpack; We also need to consider handling the.vue/.tsx;
> 
> By using the Composition API, you can avoid this issue. Would you be open to giving it a try?

How to use composite APIs to avoid this problem? Can you provide an example? thanks
